### DISCUSSION TITLE
Cannot Query IP on ASN Module

### DISCUSSION THREAD

**Discussion by mcmillster** [rspamd/rspamd#4235] [Q&A]
I've been trying to investigate an issue for some time and am curious if anyone is able to provide some insight into this issue. It appears my ASN module in RSPAMD is unable to query for results. I couldn't find much information on the subject in the docs. I replaced all the numbers below with '0' not knowing what this module is really doing besides the limited info in the docs on it.

"lua; asn.lua:78: cannot query ip 0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.b.0.0.0.0.0.0.asn6.rspamd.com on 127.0.0.53: no results"

Thanks for any insight anyone else is able to add here.

**Comment by vstakhov** [comment]
The log message is definitely bogus and I have fixed it. The underlying reason why a specific network is not listed in the Rspamd ASN data is not something that I can answer without the exact IP address I'm afraid. In general, Rspamd takes data from the BGP dumps and parses those to extract data. Presumably, there is a new source of data that is missed by this script - I don't know.

**Reply by mcmillster** [reply]
Great to know thank you! I can send you the IP privately perhaps if you are interested in taking a look. However, I see this error on every email scanned in my logs.

Appreciate the quick response

**Reply by mcmillster** [reply]
More specifically what started this investigation is I get a "HFILTER_HOSTNAME_UNKNOWN" with 2.5 points added to every email checked.

**Reply by vstakhov** [reply]
I would appreciate if you send me some examples of the problematic addresses so I could check them. Do you see this message for IPv4 addresses?

**Reply by mcmillster** [reply]
Appreciate it! Emailed them directly to you. I don't see any logged IPv4 addresses.

**Reply by vstakhov** [reply]
Ok, it took me some time to dig into this issue. Unfortunately, there is a clash in ARIN IPv6 allocation between /24 and /25 IPv6 networks. BGP routes announce /25 route whilst ARIN announces /24 in the whois lookup for a specific network. So I guess you have found a rare case where routing information is beyond of the real addresses allocation procedure. There is nothing I can do from my side.

**Reply by mcmillster** [reply]
Understood, really appreciate you looking into the issue. Would you have some additional documentation that outlines the process of what is happening with ASN/what's being sent? Perhaps I can build a different endpoint then the one you spun up so I can utilize /24? If so I'm happy to expose this for others to use. My understanding of this module is quite lacking at the moment.
